tests/inference-mock/app.py

```diff
@@ -0,0 +1,63 @@
+from flask import Flask, request, Response, jsonify
+import logging
+
+from qna.qna import MockFactGenerator
+
+# Globals
+app = Flask(__name__)
+port_number = 11434
+app.logger.setLevel(logging.DEBUG)
+fact_generator = MockFactGenerator(app.logger)
+
+def completion_response(content:str, model:str="gpt-3.5") -> dict:
+    response = {
+        "id": "chatcmpl-2nYZXNHxx1PeK1u8xXcE1Fqr1U6Ve",
+        "object": "chat.completion",
+        "created": "12345678",
+        "model": model,
+        "system_fingerprint": "fp_44709d6fcb",
+        "choices": [
+            {
+                "text": content,
+                "content": content,
+                "index": 0,
+                "logprobs": None,
+                "finish_reason": "length"
+            },
+        ],
+        "usage": {
+            "prompt_tokens": 10,
+            "completion_tokens": 20,
+            "total_tokens":30,
+        }
+    }
+
+    return response
+
+# Routes
+@app.route('/v1/completions', methods=['POST'])
+def mock_generate():
+    if not request.json or 'prompt' not in request.json:
+        app.logger.debug("prompt is empty or None")
+        return Response("bad request", 500)
+
+    prompt_debug_str = request.json['prompt'][:90] + "..."
+    app.logger.debug(f"{request.method} {request.url} {request.json['model']} {prompt_debug_str}")
+    if request.json == None:
+        return Response("Bad Request: request json empty", status=404)
+    
+    prompt = request.json['prompt']
+    
+    # handle prompt and generate correct response
+    completion = fact_generator.generatePair(prompt)
+    if not completion:
+        completion = ""
+
+    response = jsonify(completion_response(completion, request.json['model']))
+
+    app.logger.debug(f"response: {response}")
+    return response
+    
+
+if __name__ == '__main__':
+    app.run(debug=True, port=port_number)
```

Comment:
I like this mock small app! Nice idea to save some resources doing this live.

Reply:
Thanks! I was hoping other teams could get some value out of this along with the other work in this PR as we move towards notebooks as a core part of our work.

Reply:
Yes! I was thinking if this catches on, it could even be its own separate library.

Reply:
Happy to move it into the `ci-actions` repo if thats a good home for it